testing/trino-testing/src/main/java/io/trino/testing/PlanDeterminismChecker.java

```diff
@@ -49,13 +48,15 @@ public void checkPlanIsDeterministic(String sql)
 
     public void checkPlanIsDeterministic(Session session, String sql)
     {
-        IntStream.range(1, MINIMUM_SUBSEQUENT_SAME_PLANS)
-                .mapToObj(attempt -> getPlanText(session, sql))
-                .map(planEquivalenceFunction)
-                .reduce((previous, current) -> {
```

Comment:
You can keep it the stream way, just replace `reduce` with `forEach`

Reply:
Well, this was coded using `reduce` because it depends on access to adjacent elements in the stream (to make sure that all elements are equal). With `forEach` the handler would have to keep state between invocations. @kokosing suggested to turn it into a loop, and I too think that when we rely on side effects the loop way is probably a better idea anyway.

Reply:
I missed that part. You're right.